src/garage/_dtypes.py

```diff
@@ -492,3 +492,225 @@ def __new__(cls, env_spec, observation, action, reward, next_observation,
         return super().__new__(TimeStep, env_spec, observation, action, reward,
                                next_observation, terminal, env_info,
                                agent_info)
+
+
+class TaskEmbeddingTrajectoryBatch(
```

Comment:
this is definitely the wrong way to do this. task embeddings should just stuff/unpack this stuff from env_infos

Reply:
Yeah, I agree. It's preferable not to create a bunch of new `TrajectoryBatch` "subtypes."

Reply:
An alternative compromise would be to add an optional field to `TrajectoryBatch` called `worker_infos` or something, to avoid potentially conflicting with existing `env_infos`.